src/Controller/CartResourceController.php

```diff
@@ -217,48 +187,27 @@ public function clearItems(OrderInterface $cart) {
   }
 
   public function addItems(Request $request) {
-    try {
-      $received = (string) $request->getContent();
-      $document = $this->serializer->decode($received, 'api_json');
-    }
-    catch (UnexpectedValueException $e) {
-      // If an exception was thrown at this stage, there was a problem decoding
-      // the data. Throw a 400 HTTP exception.
-      throw new BadRequestHttpException($e->getMessage());
-    }
-    if (empty($document['data'])) {
-      throw new BadRequestHttpException('Document must contain data');
-    }
-
-    // Do an initial validation of the payload before any processing.
-    foreach ($document['data'] as $key => $order_item_data) {
-      if (!isset($order_item_data['purchased_entity_type'])) {
-        throw new UnprocessableEntityHttpException(sprintf('You must specify a purchasable entity type for row: %s', $key));
-      }
-      if (!isset($order_item_data['purchased_entity_id'])) {
-        throw new UnprocessableEntityHttpException(sprintf('You must specify a purchasable entity ID for row: %s', $key));
-      }
-      if (!$this->entityTypeManager->hasDefinition($order_item_data['purchased_entity_type'])) {
-        throw new UnprocessableEntityHttpException(sprintf('You must specify a valid purchasable entity type for row: %s', $key));
-      }
-    }
+    $resource_type = new ResourceType('virtual_cart', 'virtual_cart', EntityInterface::class);
```

Comment:
Oh, wait, could we do this and pass `OrderInterface::class` to work around the bundle constraints of JSON API?

Reply:
I don't really know where you're going with this